### ISSUE TITLE
Fail on Vagrant up: Warning: Config file /etc/puppet/hiera.yaml not found, using Hiera defaults

### ISSUE THREAD

**Issue by Baadier-Sydow** [ChrisWiegman/Primary-Vagrant#23]
This is the full dump with some of the domains edited out:

```
C:\Dev\Environments\WordPress (master)
λ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'ubuntu/trusty64' is up to date...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 => 2222 (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Remote connection disconnect. Retrying...
==> default: Machine booted and ready!
GuestAdditions 4.3.20 running --- OK.
==> default: Checking for guest additions in VM...
==> default: Checking for host entries
==> default: Configuring and enabling network interfaces...
==> default: Mounting shared folders...
    default: /vagrant => C:/Dev/Environments/WordPress
    default: /var/www/pv => C:/Dev/Environments/WordPress/sites/default
    default: /var/xdebug => C:/Dev/Environments/WordPress/xdebug
    default: /var/lib/mysql => C:/Dev/Environments/WordPress/mysql
    default: /etc/apache2/ssl => C:/Dev/Environments/WordPress/ssl
    default: /var/vagrant/conf => C:/Dev/Environments/WordPress/conf
    default: /var/www/mysite.dev => C:/Dev/Environments/WordPress/sites/mysite.dev
    default: /var/www/webgrind.pv => C:/Dev/Environments/WordPress/sites/webgrind
    default: /var/www/replacedb.pv => C:/Dev/Environments/WordPress/sites/Search-Replace-DB
    default: /var/www/mysite.dev => C:/Dev/Environments/WordPress/sites/mysite.dev
    default: /var/www/phpmyadmin.pv => C:/Dev/Environments/WordPress/sites/phpmyadmin
    default: /var/www/mysite.dev => C:/Dev/Environments/WordPress/sites/mysite.dev
    default: /var/www/wordpress.core.pv => C:/Dev/Environments/WordPress/sites/wordpress/core
    default: /var/www/wordpress.trunk.pv => C:/Dev/Environments/WordPress/sites/wordpress/trunk
    default: /var/www/wordpress.legacy.pv => C:/Dev/Environments/WordPress/sites/wordpress/legacy
    default: /var/www/wordpress.stable.pv => C:/Dev/Environments/WordPress/sites/wordpress/stable
    default: /tmp/vagrant-puppet-3/manifests => C:/Dev/Environments/WordPress/manifests
    default: /tmp/vagrant-puppet-3/modules-0 => C:/Dev/Environments/WordPress/modules
    default: /var/www/wordpress.trunk.pv/htdocs/content => C:/Dev/Environments/WordPress/sites/wordpress/content
    default: /var/www/wordpress.legacy.pv/htdocs/content => C:/Dev/Environments/WordPress/sites/wordpress/content
    default: /var/www/wordpress.stable.pv/htdocs/content => C:/Dev/Environments/WordPress/sites/wordpress/content
==> default: Machine already provisioned. Run `vagrant provision` or use the `--provision`
==> default: to force provisioning. Provisioners marked to run always will still run.
==> default: Running provisioner: puppet...
==> default: Running Puppet with init.pp...
==> default: stdin: is not a tty
==> default: Warning: Config file /etc/puppet/hiera.yaml not found, using Hiera defaults
==> default: Notice: Compiled catalog for vagrant-ubuntu-trusty-64 in environment production in 10.75 seconds
==> default: Error: Could not prefetch mysql_database provider 'mysql': Execution of '/usr/bin/mysql -NBe show databases' returned 1: ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
==> default:
==> default: Error: Execution of '/usr/bin/mysql -NBe create database if not exists `mysite` character set `utf8` collate `utf8_swedish_ci`' returned 1: ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
==> default:
==> default: Error: /Stage[main]/Main/Mysql_database[mysite]/ensure: change from absent to present failed: Execution of '/usr/bin/mysql -NBe create database if not exists `mysite` character set `utf8` collate `utf8_swedish_ci`' returned 1: ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
==> default:
==> default: Error: Could not set 'link' on ensure: Protocol error - (/var/www/phpmyadmin.pv/config.inc.php, /var/www/phpmyadmin.pv/phpmyadmin/config.inc.php) at 265:/tmp/vagrant-puppet-3/manifests/init.pp
==> default: Error: Could not set 'link' on ensure: Protocol error - (/var/www/phpmyadmin.pv/config.inc.php, /var/www/phpmyadmin.pv/phpmyadmin/config.inc.php) at 265:/tmp/vagrant-puppet-3/manifests/init.pp
==> default: Wrapped exception:
==> default: Protocol error - (/var/www/phpmyadmin.pv/config.inc.php, /var/www/phpmyadmin.pv/phpmyadmin/config.inc.php)
==> default: Error: /Stage[main]/Main/File[/var/www/phpmyadmin.pv/phpmyadmin/config.inc.php]/ensure: change from absent to link failed: Could not set 'link' on ensure: Protocol error - (/var/www/phpmyadmin.pv/config.inc.php, /var/www/phpmyadmin.pv/phpmyadmin/config.inc.php) at 265:/tmp/vagrant-puppet-3/manifests/init.pp
==> default: Error: Execution of '/usr/bin/mysql -NBe create database if not exists `mysite` character set `utf8` collate `utf8_swedish_ci`' returned 1: ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
==> default:
==> default: Error: /Stage[main]/Main/Mysql_database[mysite]/ensure: change from absent to present failed: Execution of '/usr/bin/mysql -NBe create database if not exists `mysite` character set `utf8` collate `utf8_swedish_ci`' returned 1: ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
==> default:
==> default: Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: executed successfully
==> default: Notice: /Stage[main]/Composer/Exec[composer-update]/returns: executed successfully
==> default: Notice: /Stage[main]/Mailcatcher::Service/Service[mailcatcher]/ensure: ensure changed 'stopped' to 'running'
==> default: Notice: /Stage[main]/Mysql::Server::Service/Service[mysqld]/ensure: ensure changed 'stopped' to 'running'
==> default: Notice: /Stage[main]/Main/Mysql_database[wordpress.legacy.pv]/ensure: created
==> default: Notice: /Stage[main]/Main/Mysql_database[wordpress.stable.pv]/ensure: created
==> default: Notice: /Stage[main]/Main/Mysql_database[wordpress.trunk.pv]/ensure: created
==> default: Notice: /Stage[main]/Main/Mysql_database[wordpress.core.pv]/ensure: created
==> default: Notice: /Stage[main]/Main/Mysql_database[mysite]/ensure: created
==> default: Notice: /Stage[main]/Apache/Service[apache]/ensure: ensure changed 'stopped' to 'running'
==> default: Notice: Finished catalog run in 32.12 seconds
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

C:\Dev\Environments\WordPress (master)
λ vagrant provision
==> default: Running provisioner: puppet...
==> default: Running Puppet with init.pp...
==> default: stdin: is not a tty
==> default: Warning: Config file /etc/puppet/hiera.yaml not found, using Hiera defaults
==> default: Notice: Compiled catalog for vagrant-ubuntu-trusty-64 in environment production in 8.13 seconds
==> default: Error: Could not set 'link' on ensure: Protocol error - (/var/www/phpmyadmin.pv/config.inc.php, /var/www/phpmyadmin.pv/phpmyadmin/config.inc.php) at 265:/tmp/vagrant-puppet-3/manifests/init.pp
==> default: Error: Could not set 'link' on ensure: Protocol error - (/var/www/phpmyadmin.pv/config.inc.php, /var/www/phpmyadmin.pv/phpmyadmin/config.inc.php) at 265:/tmp/vagrant-puppet-3/manifests/init.pp
==> default: Wrapped exception:
==> default: Protocol error - (/var/www/phpmyadmin.pv/config.inc.php, /var/www/phpmyadmin.pv/phpmyadmin/config.inc.php)
==> default: Error: /Stage[main]/Main/File[/var/www/phpmyadmin.pv/phpmyadmin/config.inc.php]/ensure: change from absent to link failed: Could not set 'link' on ensure: Protocol error - (/var/www/phpmyadmin.pv/config.inc.php, /var/www/phpmyadmin.pv/phpmyadmin/config.inc.php) at 265:/tmp/vagrant-puppet-3/manifests/init.pp
==> default: Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: executed successfully
==> default: Notice: /Stage[main]/Composer/Exec[composer-update]/returns: executed successfully
==> default: Notice: /Stage[main]/Mailcatcher::Service/Service[mailcatcher]/ensure: ensure changed 'stopped' to 'running'
==> default: Notice: Finished catalog run in 20.63 seconds
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```
One of the comments seemed to suggest that running provision after up might get it to work but it didnt work for me.

**Comment by Baadier-Sydow** [comment]
Oops sorry about the unintended open and close of the issue.

**Comment by michaelbeil** [comment]
No worries. 

What have you read through in our docs?

**Comment by Baadier-Sydow** [comment]
@michaelbeil I wasnt aware of any documentation beyond the readme at https://github.com/ChrisWiegman/Primary-Vagrant

Is there another set of documentation?

**Comment by michaelbeil** [comment]
No, just the README: https://github.com/ChrisWiegman/Primary-Vagrant/blob/master/README.md

Just making sure you have done the following:

Install [Vagrant](http://vagrantup.com), [VirtualBox](http://virtualbox.org), and the [VirtualBox extensions](https://www.virtualbox.org/wiki/Downloads) for your environment.

Once Vagrant is installed you'll want two plugins to update your local hosts file and update the VirtualBox Guest additions in the Ubuntu install.

```vagrant plugin install vagrant-vbguest```

```vagrant plugin install vagrant-hostsupdater```

You might just ```vagrant destroy``` and ```vagrant up``` again.

**Comment by Baadier-Sydow** [comment]
Im running Windows 8.1 Pro and hostsupdater isnt playing nicely so I have manually edited my hosts file via Safe Mode then commented out those entries in my Vagrant file:

This was working on my previous Primary-vagrant set up without issues and then I got the same error as I'm currently reporting and tried to do a fresh install.

```
# Note: that much of the documentation in this file is from Varying Vagrant Vagrants, the original base for this project

Vagrant.configure("2") do |config|
  
  # Store the current version of Vagrant for use in conditionals when dealing
  # with possible backward compatible issues.
  vagrant_version = Vagrant::VERSION.sub(/^v/, '')

  # Default Ubuntu Box
  # 
  # This box is provided directly by Canonical and is updated almost nightly. Currently it is
  # configured to use Ubuntu 12.04 x64. For a full list of boxes provided by Canonical visit
  # http://cloud-images.ubuntu.com/vagrant/
  config.vm.box = "ubuntu/trusty64"

  #config.vm.hostname = "pv"

  # Default Box IP Address
  #
  # This is the IP address that your host will communicate to the guest through. In the
  # case of the default `192.168.56.101` that we've provided, Virtualbox will setup another
  # network adapter on your host machine with the IP `192.168.13.1` as a gateway.
  #
  # If you are already on a network using the 192.168.13.x subnet, this should be changed.
  # If you are running more than one VM through Virtualbox, different subnets should be used
  # for those as well. This includes other Vagrant boxes.
  config.vm.network :private_network, ip: "192.168.13.101"
  
  # Local Machine Hosts
  #
  # If the Vagrant plugin hostsupdater (https://github.com/cogitatio/vagrant-hostsupdater) is
  # installed, the following will automatically configure your local machine's hosts file to
  # be aware of the domains specified below. Watch the provisioning script as you may be
  # required to enter a password for Vagrant to access your hosts file.
  #if defined? VagrantPlugins::HostsUpdater
    #config.hostsupdater.aliases = [
    #    "mailcatcher.pv",
    # "phpmyadmin.pv",
    # "replacedb.pv",
    # "wordpress.stable.pv",
    # "wordpress.core.pv",
    # "wordpress.legacy.pv",
    # "wordpress.trunk.pv",
    # "webgrind.pv",
    #]
  #end

  # Forward Agent
  #
  # Enable agent forwarding on vagrant ssh commands. This allows you to use identities
  # established on the host machine inside the guest. See the manual for ssh-add
  config.ssh.forward_agent = true

  # Configurations from 1.0.x can be placed in Vagrant 1.1.x specs like the following.
  config.vm.provider :virtualbox do |v|
    v.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
    v.customize ["modifyvm", :id, "--memory", 768]
    v.customize ["modifyvm", :id, "--name", "Primary Vagrant"]
  end

  # Drive mapping
  #
  # The following config.vm.share_folder settings will map directories in your Vagrant
  # virtual machine to directories on your local machine. Once these are mapped, any
  # changes made to the files in these directories will affect both the local and virtual
  # machine versions. Think of it as two different ways to access the same file. When the
  # virtual machine is destroyed with `vagrant destroy`, your files will remain in your local
  # environment.

  # /var/www/
  # 
  # For each project you're working on map a folder to it. The first argument is the location
  # on the host computer. The second argument is the location on the guest matching. Finally the 
  # 3rd arguement is a unique ID given to each folder mapped
  config.vm.synced_folder "sites/default", "/var/www/pv", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/wordpress/stable", "/var/www/wordpress.stable.pv", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/wordpress/legacy", "/var/www/wordpress.legacy.pv", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/wordpress/trunk", "/var/www/wordpress.trunk.pv", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/wordpress/core", "/var/www/wordpress.core.pv", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/wordpress/content", "/var/www/wordpress.stable.pv/htdocs/content", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/wordpress/content", "/var/www/wordpress.trunk.pv/htdocs/content", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/wordpress/content", "/var/www/wordpress.legacy.pv/htdocs/content", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/Search-Replace-DB", "/var/www/replacedb.pv", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/phpmyadmin", "/var/www/phpmyadmin.pv", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/webgrind", "/var/www/webgrind.pv", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/mysite.dev", "/var/www/mysite.dev", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/mysite.dev", "/var/www/mysite.dev", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/mysite.dev", "/var/www/mysite.dev", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  
  # 
  # Specify a folder for various vagrant data. A MySQL data folder would be appropriate here (for example).
  config.vm.synced_folder "xdebug", "/var/xdebug", :mount_options => [ "dmode=777", "fmode=777" ]
  config.vm.synced_folder "ssl", "/etc/apache2/ssl", :mount_options => [ "dmode=777", "fmode=777" ]
  config.vm.synced_folder "conf", "/var/vagrant/conf", :mount_options => [ "dmode=777", "fmode=777" ]
  config.vm.synced_folder "mysql", "/var/lib/mysql", :mount_options => [ "dmode=777", "fmode=777" ]

  # Custom Mappings
  #
  # Use this section to specifiy custom mapptings for your own development.

  # Provisioning
  #
  # Process one or more provisioning scripts depending on the existence of custom files.
  # 
  # Provisioning uses the Puppet configuration tool (http://puppetlabs.com/). This tool
  # relies on modules in the modules/ folder which are configures in manifests/default.pp.
  config.vm.provision "puppet", run: "always" do |puppet|
    puppet.manifests_path = "manifests"
    puppet.manifest_file = "init.pp"
    puppet.module_path = "modules"
    puppet.facter = { "fqdn" => "pv" }
  end


end
```

**Comment by ptt-homme** [comment]
Hi @Baadier-Sydow,

I've never had this issue before and I was previously thinking like @michaelbeil for the mandatory steps. Anyway, I guess you followed them.

This seems to be a little bit weird:
```
  config.vm.synced_folder "sites/webgrind", "/var/www/webgrind.pv", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/mysite.dev", "/var/www/mysite.dev", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/mysite.dev", "/var/www/mysite.dev", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
  config.vm.synced_folder "sites/mysite.dev", "/var/www/mysite.dev", :owner => "www-data", :mount_options => [ "dmode=775", "fmode=774" ]
```
Why do you need it three times? Have you done some modification to custom files such as apache.pp ? Have you tried to clone the repo again, using --recursive?

Ptt-homme

**Comment by Baadier-Sydow** [comment]
Thanks for helping out @ptt-homme 

I just edited out my custom domain mappings as it was for client work and replaced it with mysite.dev.

Probably lame, but I didnt want to leave it public.

Yep on the --recursive. 

The box was running well without any issues when it started generating those errors. I then created a new clone and pulled it in with --recursive on to see if it would help but it hasnt.

Its very strange. I dont remember doing anything in particular when it stopped working.

**Comment by ptt-homme** [comment]
And what about Vagrant, Virtualbox versions?

**Comment by Baadier-Sydow** [comment]
Virtualbox: 4.3.20
Vagrant: 1.6.5

On Wed, Jan 28, 2015 at 7:59 PM, ptt-homme <notifications@github.com> wrote:

> And what about Vagrant, Virtualbox versions?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ChrisWiegman/Primary-Vagrant/issues/23#issuecomment-71882842>
> .
>



-- 


*Baadier Sydow**, Sem Zoom*
Mobile: (071) 219-8512
Office: (021) 813-9077
Skype: semzoom
Email: contact@semzoom.com
Web: http://semzoom.com


[image: Facebook Page] <http://www.facebook.com/SEMZoom> [image: Twitter]
<http://twitter.com/sem_zoom> [image: LinkedIn]
<http://www.linkedin.com/company/sem-zoom> [image: YouTube]
<http://www.youtube.com/user/SEMZoomVideo> [image: Google Plus Page]
<https://plus.google.com/u/0/108913346530622227146>

**Comment by michaelbeil** [comment]
Would you upgrade to the most recent version of vagrant? 

https://docs.vagrantup.com/v2/installation

**Comment by michaelbeil** [comment]
Currently at 1.7.2.

**Comment by Baadier-Sydow** [comment]
Thanks, I'll give it a shot.

**Comment by Baadier-Sydow** [comment]
Still no luck unfortunately.

**Comment by michaelbeil** [comment]
Did you install Vagrant as a gem or with the .dmg?

**Comment by Baadier-Sydow** [comment]
Running it on Windows 8.1 Pro. Thats probably the start of my problem.Ive been running XAMPP in the interim but its not ideal. Its really strange how it worked one day and just died the next.

**Comment by michaelbeil** [comment]
Oh gotcha. Well, I don't see why you shouldn't be able to run it on Windows. 

Same error as last time upon `vagrant up`?

**Comment by Baadier-Sydow** [comment]
Correct. No change at all.

**Comment by michaelbeil** [comment]
Ok. To sum up, you're running Windows 8.1 Pro.

You have installed Vagrant 1.7.2 64bit, Virtualbox: 4.3.20, and the VirtualBox extensions for your environment.

You have attempted to install these two plugins:

`vagrant plugin install vagrant-vbguest`

`vagrant plugin install vagrant-hostsupdater`

And `hostsupdater` failed. What was wrong with `hostsupdater`?

Have you tried the [/precise](https://github.com/ChrisWiegman/Primary-Vagrant/archive/Precise.zip) branch? 

Or try our most recent release: [1.1.2](https://github.com/ChrisWiegman/Primary-Vagrant/releases/tag/1.1.2)

**Comment by Baadier-Sydow** [comment]
hostsupdater failed because of permissions issues on the hosts file on Windows. The only way I can edit my hosts file is by logging into safe mode and then editing as admin. It just doesnt work on Windows 8.1

I havent tried the most recent Primary-Vagrant build yet. I'll give it a shot and report back.

Thanks for the help.

**Comment by michaelbeil** [comment]
Sounds good. Let me know how it goes.

**Comment by michaelbeil** [comment]
@Baadier-Sydow any updates?

**Comment by Baadier-Sydow** [comment]
Thanks for following up @michaelbeil 

I've been dealing with a backlog but have added testing it to my to-do list. I'll get back shortly with a follow up.

**Comment by michaelbeil** [comment]
I shall await your feedback.

**Comment by michaelbeil** [comment]
@Baadier-Sydow any news?

**Comment by thomasdebruin** [comment]
Same error here, on the hiera.yaml, using latest versions of Primary Vagrant & VBOX on windows 8.1 Pro The whole vagrant and puppet thing is new to me. Despite the error everything seems to work just fine (for now).

**Comment by AvalonMel** [comment]
Same error here.
VirtualBox 4.3.28
Vagrant 1.7.2
Mac OSX

Cloned with --recursive
Installed plugins

Get this error when I vagrant up
```
default: Running provisioner: puppet...
==> default: Running Puppet with init.pp...
==> default: stdin: is not a tty
==> default: Warning: Config file /etc/puppet/hiera.yaml not found, using Hiera defaults
==> default: Error: Puppet::Parser::AST::Resource failed with error ArgumentError: Could not find declared class postfix at /tmp/vagrant-puppet/manifests-a11d1078b1b1f2e3bdea27312f6ba513/init.pp:217 on node pv
==> default: Wrapped exception:
==> default: Could not find declared class postfix
==> default: Error: Puppet::Parser::AST::Resource failed with error ArgumentError: Could not find declared class postfix at /tmp/vagrant-puppet/manifests-a11d1078b1b1f2e3bdea27312f6ba513/init.pp:217 on node pv
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

**Comment by ericjsilva** [comment]
Looks like this is an error caused by the postfix missing error documented in issue #41.

As a workaround, you can remove the `/modules/postfix` folder and then from the `modules` folder run `git clone --recursive git@github.com:Aethylred/puppet-postfix.git postfix`

Once done go back to the `PV` root and run `vagrant up`

**Comment by nmccready** [comment]
As a work around I run this shell provision

puppet-ubuntu-bugs.sh
```sh
#!/bin/bash
set -e
touch /etc/puppet/hiera.yaml
#https://github.com/mitchellh/vagrant/issues/1673
#https://github.com/dm-cracked-sean-knight/packer-templates/commit/bfec91c086e3dd318d4c16b3658a9db4a5c08d15
sed -i -r -e 's/^([# ]+)?(mesg n)/# \2/' /root/.profile

#https://github.com/comperiosearch/vagrant-elk-box/issues/19
sed -e '/templatedir/ s/^#*/#/' -i.back /etc/puppet/puppet.conf

```

**Comment by michaelbeil** [comment]
This was fixed in #42.
